play/src/test/scala/net/yao/json/CollectionTest.scala

```diff
@@ -0,0 +1,150 @@
+package net.zhenglai
+package net.yao.json
+
+import org.scalatest.FunSuite
+
+import scala.collection.mutable
+
+class CollectionTest extends FunSuite {
+
+  test("immutable array") {
```

Comment:
java Array is not immutable, not recommend to use it

Reply:
yes, you are right!  After my research from internet, I still can't understand the implementation of the method `update`.
`  def update(i: Int, x: T): Unit = { throw new Error() }`

Reply:
Scala Array is just wrap of Java Array, I think it just a simple bridge and will be translated to JVM's definition of array index assignment, compiler will do this stuff.